### DISCUSSION TITLE
How to get data from the nested M2A?

### DISCUSSION THREAD

**Discussion by 4rt10m** [directus/directus#12810] [Archive]
Hey all!

I'm new to directus and data modelling in general, coming from Prismic CMS. And having a hard time getting the data I need.

I have a collection "home" which has a M2A field called "sections".
I also have a collection "providers" with a bunch of input fields for the provider info.

One of the "sections" I have is "gambling_table" (created as another collection), which has a M2A linked to the "providers" collection.

The idea is, that in the homepage, I can: select the gambling_table section, from there fill in the input data and select the "providers".

Now, when I query: `http://0.0.0.0:8055/items/home?fields=*.*.*`  (using REST)

I get all the top-level "sections" data, other than the M2A "providers" field on the gambling_table collection (where I'm only getting the ids), screenshot below:
<img width="946" alt="Screenshot 2022-04-17 at 13 27 11" src="https://user-images.githubusercontent.com/21058332/163714447-8452dddd-d066-43af-b057-ef5befa8aed2.png">

So here are the questions:
- How can I get all the data in the "providers" (not the IDs) with a single query?
- Is there any better / other way to structure this to get all the data from single query? (I've tried setting up the O2M to link the gambling_table to the providers, but that only returns the page titles, not all the data I need).
- Will it be easier to achieve with GraphQL?

P.S here is an example response from the "providers" (the data I want to see instead of the IDs):
<img width="893" alt="Screenshot 2022-04-17 at 13 39 35" src="https://user-images.githubusercontent.com/21058332/163714993-f79022fb-4e74-4f7c-8f0b-c61afcf81652.png">

Thank you all!

**Comment by benhaynes** [comment]
You can request more levels:
`http://0.0.0.0:8055/items/home?fields=*.*.*.*.*`

Or get more specific with field names (instead of wildcards) so you don't fetch too much data.

**Reply by rijkvanzanten** [reply]
For example: `fields=sections.item:gambling_table.providers.*`. See https://docs.directus.io/reference/introduction/#many-to-any-union-types for more information on the nested m2a selection

**Reply by 4rt10m** [reply]
Thank you! All works 👍🏻
